### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by Joel-100**
- Created Classroom, Book, and Rental classes
- Created relationships between Classroom - Student and Person - Book classes

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+  end
```
Comment (danmainah) [inline book.rb]: - [ ] @Joel-100 great work on your association, however, your add_rental method needs to include the date, self, and person as parameters to create a new rental

File: person.rb [hunk 2]
```diff
@@ -29,4 +29,8 @@ def can_use_services?
   def correct_name
     @name
   end
+
+  def add_rental(rental)
+    @rentals << rental
+  end
```
Comment (danmainah) [inline person.rb:35]: - [ ] @Joel-100 great work on your association, however, your add_rental method needs to include the date, self, and person as parameters to create a new rental. I also noted you have not initialized rentals which will be used in this method

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Joel-100 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: book.rb (hunk 1 above)
Comment (Joel-100) [reply book.rb]: Okay

File: person.rb (hunk 2 above)
Comment (Mihndim2020) [inline person.rb:35]: - [OPTIONAL] The `add_rental` method should be added to the person class like so:
```def add_rental(book, date)```
   ``` Rental.new(date, book, self)```
 ``` end```
Please kindly fix that.

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @Joel-100, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (Joel-100) [reply person.rb:35]: Alright noted, thanks for that!

**Pull request merged**